releasing-archway.md

```diff
@@ -0,0 +1,112 @@
+# Archway Protocol Release Versioning
+
+Reference: [Semantic Versioning](http://semver.org)
+
+Legend:
+
+- **X.Y.Z** refers to the version (git tag) of Archway Protocol that is released.
+- **Network Operator** refers to an entity running a node and/or a validator and/or a relayer.
+- **archway-1** refers to the chain id of archway protocl main network.
+- **constantine-1** refers to the chain-id of the archway protocol canary public test network.
+- **titus-1** refers to the the chain-id of the archway protocol development public test network.
+
+Note: Please refer to https://github.com/archway-network/networks for a registry of all public networks running archway protocol
+and network specific information.
+
+## Release versioning
+
+### Minor version scheme and timeline
+- X.Y.Z-rc.W (Branch: release-X.Y)
+  - When main is feature-complete for X.Y, we will cut the release-X.Y
+    branch and cherrypick only PRs essential to X.Y.
+  - If we're not satisfied with X.Y.0-rc.0, we'll release other rc releases,
+    (X.Y.0-rc.W | W > 0) as necessary.
+- X.Y.0 (Branch: release-X.Y)
+  - Final release, cut from the release-X.Y branch cut two weeks prior.
+  - X.Y.1-rc.0 will be tagged at the same commit on the same branch.
+- X.Y.Z, Z > 0 (Branch: release-X.Y) ([Patch releases](#patch-releases))
+  - [Patch releases](#patch-releases) are released as we cherrypick commits from main into
+    the release-X.Y branch, as needed.
+  - X.Y.Z is cut straight from the release-X.Y branch, and X.Y.Z+1-beta.0 is
+    tagged on the followup commit that updates pkg/version/base.go with the beta
+    version.
+- X.Y.Z, Z > 0 (Branch: release-X.Y.Z) (Branched [patch releases](#patch-releases) only for hotfix situations)
+  - These are rarely used and are special in that the X.Y.Z tag is branched to isolate
+    the emergency/critical fix from all other changes that have landed on the
+    release branch since the previous tag
+  - Cut release-X.Y.Z branch to hold the isolated patch release
+  - Tag release-X.Y.Z branch + fixes with X.Y.(Z+1)
+  - Branched [patch releases](#patch-releases) are rarely needed but used for
+    emergency/critical fixes to the latest release
+
+### Major version timeline
+
+There is currently no mandated timeline for major versions beyond version 1.Y.Z, Y,Z >= 0. We haven't so far applied a rigorous interpretation of semantic
+versioning with respect to incompatible changes of any kind.
```

Comment:
> We haven't so far applied a rigorous interpretation of semantic
versioning with respect to incompatible changes of any kind

So when I read the beginning of this proposal https://github.com/archway-network/archway/pull/303/files#diff-59d7c43d237cd15535dbbee60e1a521267108d179480a97e65986810f8fd00afR3
 the first question that came to my mind: What does semantic versioning mean in the context of a full blockchain node? How about hard forks? 

If we delay the "full" semantics, as stated above, I was wondering if this very proposal can be simplified further ...

Reply:
Correct. Semantic versioning is a reference standard for this proposal and this proposal does not aim to implement "full" semantic versioning standard.

>  What does semantic versioning mean in the context of a full blockchain node? How about hard forks?

I am not sure I understand the question. Please feel free to suggest any changes which might help us further simplify the proposal

Reply:
A hard fork is in a sense a "breaking change" for the blockchain but it is not necessarily a breaking change for the code, that is the code can be perfectly backwards compatible. So my question is about the need to clarify this distinction: when we use the word "semantic", do we mean just the code, or the resulting blockchain as a whole?

I hope that helps.

Reply:
> when we use the word "semantic", do we mean just the code, or the resulting blockchain as a whole

This proposal only deals with release and versioning of the software itself and that in turn dictates how the networks consume that software. I will update the doc to make the distinction clear

Reply:
I am still confused as to our interpretation of "semantic versioning" here. It is my general understanding that you have two distinct versions, namely a `build` version and a `release` version, once is for code, developer and API tracking (space where semver applies) and the other for human / marketing consumption (e.g. Windows 8)...

As an example we could conceivable have a scenario where the fundamental underlying functionality of the network remains unchanged, despite an upstream dependency requiring a state-breaking change in order to integrate a critical security patch. Thus we would still have archway-network-v1 (release version) and our build version should progress from v1.2.3 to v2.0.0 in order to indicate this breaking change (as per semver definition).

Reply:
As per our discussion the runtime build and API release versions are to be treated as two separate matters;

This proposal is focused on the runtime build versioning. X.Y.Z <- X bumped on any state breaking changes regardless of current API release version;

Reply:
Added distinctions on the top of the document